server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java

```diff
@@ -605,6 +609,11 @@ public OAuth2AccessToken createAccessToken(OAuth2Authentication authentication)
             additionalRootClaims = new HashMap<>(uaaTokenEnhancer.enhance(emptyMap(), authentication));
         }
 
+        String clientAuthentication = ofNullable(oAuth2Request.getExtensions().get(CLIENT_AUTH_METHOD)).map(String.class::cast).orElse(null);
```

Comment:
This is in a very long method - please move the new code to a new method. Probably two methods, because clientAuthentication is used later.

Reply:
I tried to to, but then the logic is more complex because I need both values, clientAuthentication and additionaClaims. Extracting this into method is therefore not possible and other addtional call / actions are bigger then these 4 lines